sdk/storage/azure-storage-blob/tests/test_common_blob_async.py

```diff
@@ -3098,4 +3196,21 @@ async def test_validate_empty_blob(self, **kwargs):
         assert await blob_client.exists()
         assert (await blob_client.get_blob_properties()).size == 0
 
+    @BlobPreparer()
+    @recorded_by_proxy_async
+    async def test_set_blob_tier_cold_tier(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+
+        await self._setup(storage_account_name, storage_account_key)
+        blob_name = await self._create_block_blob(standard_blob_tier=StandardBlobTier.Hot, overwrite=True)
+        blob = self.bsc.get_blob_client(self.container_name, blob_name)
+        await blob.set_standard_blob_tier(StandardBlobTier.Cold)
+
+        # Act
+        props = await blob.get_blob_properties()
+
+        # Assert
+        assert props.blob_tier == StandardBlobTier.Cold
+
```

Comment:
Not sure where to put this comment but let's also a test for `list_blobs` probably in `test_container` since it's a new property there too. 

If there is an existing test that tests access tier, maybe just add a Cold blob into that test instead of making a whole new test. If there is not a test, maybe test Hot, Cool, and Cold in one test for completeness.

Reply:
Sounds good! For now, I made them standalone tests as the only tests in `test_container` that assert against the tier are live-only, so I think for simplicity I will leave them standalone (for both `list_blobs` and `walk_blobs`)